narwhal/types/src/primary.rs

```diff
@@ -802,3 +864,32 @@ pub struct RoundVoteDigestPair {
     /// The hash of the vote used to ensure equality
     pub vote_digest: VoteDigest,
 }
+
+#[cfg(test)]
+mod tests {
```

Comment:
Question for my own edification, I'm curious if we have any kind of general philosphy on when to put tests in a separate file vs. not? I notice some of the tests in this crate are in separate files (e.g. `error_test.rs`) and some not.

Reply:
Ah I am mostly following the Rust guidance , where it is suggested the unit tests to be put in the same file. Well the main driver here was also that this tests were quite short so also didn't think it worth putting in separate file